### PR TITLE
entimport: Add field exclusion and error readability improvements

### PR DESCRIPTION
**Pull request by tmc**
This adds the ability to exclude a set of tables and improves error messages to indicate which table and/or column had an issue.

### CODE REVIEW

**Comment by tmc** [comment]
Wups I had a `go.work` file in place for some schemast changes I was working on -- removed reference.

File: internal/entimport/postgres.go [hunk 1]
```diff
@@ -66,7 +80,7 @@ func (p *Postgres) field(column *schema.Column) (f ent.Field, err error) {
 	case *postgres.UUIDType:
 		f = field.UUID(name, uuid.New())
 	default:
-		return nil, fmt.Errorf("entimport: unsupported type %q", typ)
+		return nil, fmt.Errorf("column %v: unsupported type %q", column.Name, typ)
```
Comment (giautm) [inline internal/entimport/postgres.go:83]:
```suggestion
		return nil, fmt.Errorf("entimport: unsupported type %q for column %v", typ, column.Name)
```

File: internal/entimport/mysql.go [hunk 2]
```diff
@@ -67,7 +81,7 @@ func (m *MySQL) field(column *schema.Column) (f ent.Field, err error) {
 	case *schema.TimeType:
 		f = field.Time(name)
 	default:
-		return nil, fmt.Errorf("entimport: unsupported type %q", typ)
+		return nil, fmt.Errorf("column %v: unsupported type %q", column.Name, typ)
```
Comment (giautm) [inline internal/entimport/mysql.go:84]:
```suggestion
		return nil, fmt.Errorf("entimport: unsupported type %q for column %v", typ, column.Name)
```

File: internal/entimport/import.go [hunk 3]
```diff
@@ -324,7 +335,7 @@ func schemaMutations(field fieldFunc, tables []*schema.Table) ([]schemast.Mutato
 		}
 		node, err := upsertNode(field, table)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("issue with table %v: %w", table.Name, err)
```
Comment (giautm) [inline internal/entimport/import.go:338]:
```suggestion
			return nil, fmt.Errorf("entimport: issue with table %v: %w", table.Name, err)
```

File: internal/entimport/mysql.go (hunk 2 above)
Comment (tmc) [reply internal/entimport/mysql.go:85]: I pulled these because this gets wrapped higher in the stack.

**Comment by tmc** [comment]
I accepted those suggestions but note that it will render errors such as: `2022/07/19 08:46:03 entimport: schema import failed - entimport: issue with table foo_bar_baz: entimport: invalid primary key - single part key must be present (table: foo_bar_baz, got: 3 parts)`

**Comment by giautm** [comment]
> I accepted those suggestions but note that it will render errors such as: `2022/07/19 08:46:03 entimport: schema import failed - entimport: issue with table foo_bar_baz: entimport: invalid primary key - single part key must be present (table: foo_bar_baz, got: 3 parts)`

I think we should avoid to use `-` here. I will sent another PR to update the first format:

`entimport: schema import failed - %v` -> `entimport: schema import failed: %v`

File: internal/entimport/mysql.go [hunk 4]
```diff
@@ -42,7 +42,22 @@ func (m *MySQL) SchemaMutations(ctx context.Context) ([]schemast.Mutator, error)
...
+		excludedTableNames := make(map[string]bool)
+		for _, t := range m.excludedTables {
+			excludedTableNames[t] = true
+		}
+		// filter out tables that are in excludedTables:
+		for _, t := range s.Tables {
+			if !excludedTableNames[t.Name] {
+				tables = append(tables, t)
+			} else {
+			}
```
Comment (giautm) [inline internal/entimport/mysql.go:57]: Do you see the lint issue here?
```suggestion
			}
```

File: internal/entimport/postgres.go [hunk 5]
```diff
@@ -37,7 +37,22 @@ func (p *Postgres) SchemaMutations(ctx context.Context) ([]schemast.Mutator, err
...
+		excludedTableNames := make(map[string]bool)
+		for _, t := range p.excludedTables {
+			excludedTableNames[t] = true
+		}
+		// filter out tables that are in excludedTables:
+		for _, t := range s.Tables {
+			if !excludedTableNames[t.Name] {
+				tables = append(tables, t)
+			} else {
+			}
```
Comment (giautm) [inline internal/entimport/postgres.go:52]:
```suggestion
			}
```

**Pull request merged**